plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/AppEngineStandardFacet.java

```diff
@@ -116,13 +118,20 @@ public static boolean isAppEngineStandardRuntime(
    */
   public static void installAppEngineFacet(IFacetedProject facetedProject,
       boolean installDependentFacets, IProgressMonitor monitor) throws CoreException {
-    
+
     SubMonitor subMonitor = SubMonitor.convert(monitor, 100);
-    
+
+    // https://github.com/GoogleCloudPlatform/google-cloud-eclipse/issues/1155
+    // Instead of using "IFacetedProject.installProjectFacet()", we install facets in a batch
+    // using "IFacetedProject.modify()" so that we hold the lock until we finish installing
+    // all the facets. This ensures that the first ConvertJob starts installing the JSDT facet only
+    // after the batch is complete, which in turn prevents the first ConvertJob from scheduling
+    // the second ConvertJob (triggered by installing the JSDT facet.)
+    Set<IFacetedProject.Action> facetInstallBatchSet = new HashSet<>();
```

Comment:
I think  just facetInstallBatch, or facetInstallSet; or is there some meaning of "Batch" separate from set here?